src/datatypes/interval.coffee

```diff
@@ -291,17 +292,46 @@ module.exports.Interval = class Interval
     startGreaterThanOrEqual && endEqual
 
   width: () ->
-    if @low instanceof DateTime or @high instanceof DateTime
-      throw new Error("Width of DateTime intervals is not supported")
+    if (@low? and (@low.isDateTime or @low.isDate or @low.isTime)) or
+       (@high? and (@high.isDateTime or @high.isDate or @high.isTime))
+      throw new Error("Width of Date, DateTime, and Time intervals is not supported")
 
     closed = @toClosed()
     if closed.low instanceof Uncertainty or closed.high instanceof Uncertainty
       null
+    else if closed.low.isQuantity
+      if closed.low.unit != closed.high.unit
+        throw new Error("Cannot calculate width of Quantity Interval with different units")
+      lowValue = closed.low.value
+      highValue = closed.high.value
+      diff = Math.abs(highValue - lowValue)
+      Math.round(diff * Math.pow(10, 8)) / Math.pow(10, 8)
+      new Quantity(value: diff, unit: closed.low.unit)
     else
       # TODO: Fix precision to 8 decimals in other places that return numbers
       diff = Math.abs(closed.high - closed.low)
       Math.round(diff * Math.pow(10, 8)) / Math.pow(10, 8)
 
+  size: (pointSize) ->
```

Comment:
The pointSize is calculated by `getPointSize` in `src/elm/interval.coffee`

Your comment made me realize this would be better placed in `src/datatypes/interval.coffee`, so I refactored it.

Now it is called as `interval.getPointSize()`